test/arrayops.jl

```diff
@@ -2059,12 +2059,12 @@ struct F21666{T <: Base.TypeArithmetic}
     x::Float32
 end
 
+Base.TypeArithmetic(::Type{F21666{T}}) where {T} = T()
```

Comment:
Might make more sense to expand the Methods/Redefining Methods manual section a bit to explain how things behave inside `let`/`try`/... blocks at top level and add a warning note to the Unit Testing/Working with Test Sets section. If we just add a comment here, that won't stop the same mistake from being made elsewhere. (Assuming proper documentation will is pretty optimistic, I know, but still...)